packages/sanity/src/desk/comments/src/components/list/CommentsListItemLayout.tsx

```diff
@@ -236,23 +237,32 @@ export function CommentsListItemLayout(props: CommentsListItemLayoutProps) {
     setIsEditing((v) => !v)
   }, [])
 
-  useDidUpdate(isEditing, () => {
-    setMenuOpen(false)
-  })
-
-  useGlobalKeyDown((event) => {
-    if (event.key === 'Escape' && !hasChanges) {
-      cancelEdit()
-    }
-  })
+  useDidUpdate(
+    isEditing,
+    useCallback(() => {
+      setMenuOpen(false)
+    }, []),
+  )
 
-  useClickOutside(() => {
-    if (!hasChanges) {
-      cancelEdit()
-    }
-  }, [rootElement])
+  useGlobalKeyDown(
+    useCallback(
+      (event) => {
+        if (event.key === 'Escape' && !hasChanges) {
+          cancelEdit()
+        }
+      },
+      [cancelEdit, hasChanges],
+    ),
+  )
 
-  const avatar = <CommentsAvatar user={user} />
+  useClickOutside(
+    useCallback(() => {
+      if (!hasChanges) {
+        cancelEdit()
+      }
+    }, [cancelEdit, hasChanges]),
+    [rootElement],
+  )
```

Comment:
**question:** WDYT about pulling out these 'inline' `useCallback` cases and giving them named functions (`handleXXX`)? 

I think it can better communicate intent and makes the file easier to parse – it's nice being able to get an overview of all event callbacks associated with this component in one place, and makes it easier to identify opportunities to make this DRYer, too.

**For example:**

```
  useDidUpdate(
    isEditing,
    useCallback(() => {
      setMenuOpen(false)
    }, []),
  )
```

could be replaced with

```
  useDidUpdate(isEditing, handleMenuClose)
```

Reply:
Totally agree – I can tix this!